source/client/output_transform_main.cc

```diff
@@ -0,0 +1,69 @@
+#include "client/output_transform_main.h"
+
+#include "nighthawk/common/exception.h"
+
+#include "external/envoy/source/common/protobuf/message_validator_impl.h"
+
+#include "api/client/service.pb.h"
+
+#include "common/utility.h"
+
+#include "client/factories_impl.h"
+#include "client/options_impl.h"
+#include "client/output_collector_impl.h"
+#include "client/output_formatter_impl.h"
+
+#include "absl/strings/strip.h"
+#include "fmt/ranges.h"
+#include "tclap/CmdLine.h"
+
+namespace Nighthawk {
+namespace Client {
+
+OutputTransformMain::OutputTransformMain(int argc, const char* const* argv, std::istream& input)
+    : input_(input) {
+  const char* descr = "L7 (HTTP/HTTPS/HTTP2) performance characterization transformation tool.";
+  TCLAP::CmdLine cmd(descr, ' ', "PoC"); // NOLINT
+  std::vector<std::string> output_formats = OutputFormatterImpl::getLowerCaseOutputFormats();
+  TCLAP::ValuesConstraint<std::string> output_formats_allowed(output_formats);
+  TCLAP::ValueArg<std::string> output_format(
+      "", "output-format", fmt::format("Output format. Possible values: {}.", output_formats), true,
+      "", &output_formats_allowed, cmd);
+  Utility::parseCommand(cmd, argc, argv);
+  output_format_ = output_format.getValue();
+}
+
+std::string OutputTransformMain::readInput() {
+  std::stringstream input;
+  std::string line;
+  while (getline(input_, line)) {
+    input << line << std::endl;
+  }
+  return input.str();
```

Comment:
There has to be a better way to do this in C++ I think, see how Envoy does it in https://github.com/envoyproxy/envoy/blob/master/source/common/filesystem/posix/filesystem_impl.cc#L92.

Reply:
adjusted to use `rdbuf()`